src/main/java/org/broadinstitute/hellbender/tools/spark/pathseq/PSBwaUtils.java

```diff
@@ -0,0 +1,66 @@
+package org.broadinstitute.hellbender.tools.spark.pathseq;
+
+import com.google.cloud.dataflow.sdk.options.PipelineOptions;
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMSequenceDictionary;
+import htsjdk.samtools.SAMSequenceRecord;
+import org.apache.spark.api.java.JavaRDD;
+import org.broadinstitute.hellbender.engine.datasources.ReferenceMultiSource;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.SerializableFunction;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.util.*;
+
+/**
+ * Utility functions for PathSeq Bwa tool
+ */
+public final class PSBwaUtils {
```

Comment:
I assume that extracting only referenced ref sequences is essential because you have so many of them.  Nonetheless, aren't you going to end up with a bunch of BAMs with inconsistent headers?
Sounds really complicated unless it's essential.